deltachat-jsonrpc/src/api/types/message.rs

```diff
@@ -305,6 +308,61 @@ impl From<download::DownloadState> for DownloadState {
     }
 }
 
+#[derive(Serialize, TypeDef)]
+pub enum SystemMessageType {
+    Unknown,
+    GroupNameChanged,
+    GroupImageChanged,
+    MemberAddedToGroup,
+    MemberRemovedFromGroup,
+    AutocryptSetupMessage,
+    SecurejoinMessage,
+    LocationStreamingEnabled,
+    LocationOnly,
+
+    /// Chat ephemeral message timer is changed.
+    EphemeralTimerChanged,
+
+    // Chat protection state changed
+    ChatProtectionEnabled,
+    ChatProtectionDisabled,
+
+    /// Self-sent-message that contains only json used for multi-device-sync;
+    /// if possible, we attach that to other messages as for locations.
+    MultiDeviceSync,
+
+    // Sync message that contains a json payload
+    // sent to the other webxdc instances
+    // These messages are not shown in the chat.
+    WebxdcStatusUpdate,
```

Comment:
same here - unless the file is generated, i'd hide these implementation details from exposed api.

Reply:
so we should convert them to Unknown?

Reply:
> so we should convert them to the Unknown variant?

no. if removal means adding code, better leave it as is. having "unknown" differently in UI and in core leads to confusion sooner or later.